detox/android/detox/src/main/java/com/wix/detox/espresso/ReactNativeTimersIdlingResource.java

```diff
@@ -144,7 +137,46 @@ public void doFrame(long frameTimeNanos) {
         isIdleNow();
     }
 
-    public void stop() {
-        stopped.set(true);
+    public void pause() {
+        paused.set(true);
+        if (callback != null) {
+            callback.onTransitionToIdle();
+        }
+    }
+    public void resume() {
+        paused.set(false);
+    }
+
+    private boolean isTimerOutsideBusyWindow(Object nextTimer) {
+        final long currentTimeMS = System.nanoTime() / 1000000L;
+        final Reflect nextTimerReflected = Reflect.on(nextTimer);
+        final long targetTimeMS = nextTimerReflected.field(TIMER_FIELD_TARGET_TIME).get();
+        final int intervalMS = nextTimerReflected.field(TIMER_FIELD_INTERVAL).get();
+        final boolean isRepetitive = nextTimerReflected.field(TIMER_FIELD_REPETITIVE).get();
+
+//        Log.i(LOG_TAG, "Next timer has duration of: " + intervalMS
```

Comment:
Not sure those comments are necessary, unless there's a hack that needs an explanation.

Reply:
Assuming you're referring to the comments and not the logs -
they're necessary because we didn't unit test, so there's nothing here to explain the idea behind the implementation. This will change soon, of course.